c++/src/io/InputStream.cc

```diff
@@ -56,6 +56,13 @@ namespace orc {
     // PASS
   }
 
+  void SeekableInputStream::throwFrom(const char* class_name) {
+    std::stringstream msg;
+    msg << "Read past end of stream in " << class_name
+        << " or some errors occured: " << this->getName();
+    throw ParseError(msg.str());
```

Comment:
Either rename it to throwParseErrorFrom or make it more general to throw other exceptions